### PR TITLE
Update readme, fixes #143

### LINKED ISSUE #143: Undef StanODEData unless CmdStan required pre DiffEqBayes
`StanODEData` is undefined unless `CmdStan` is required before `DiffEqBayes`. Also it is not exported so the example in the README doesn't work which might be confusing for newcomers.

This shows the problems:
```
feldt:~$ julia
...
julia> using DiffEqBayes

julia> DiffEqBayes.StanODEData()
ERROR: UndefVarError: StanODEData not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./Base.jl:13
 [2] top-level scope at REPL[3]:1

julia> exit()
feldt:~$ julia
...
julia> using CmdStan

julia> using DiffEqBayes

julia> StanODEData()
ERROR: UndefVarError: StanODEData not defined
Stacktrace:
 [1] top-level scope at REPL[3]:1

julia> DiffEqBayes.StanODEData()
DiffEqBayes.StanODEData()

julia> versioninfo()
Julia Version 1.3.1
Commit 2d5741174c (2019-12-30 21:36 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.6.0)
  CPU: Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, broadwell)
```

### PR DESCRIPTION
**Pull request by abhigupta768**
@Vaibhavdixit02

### CODE REVIEW

File: src/stan_inference.jl [hunk 1]
```diff
@@ -50,7 +50,7 @@ function stan_inference(prob::DiffEqBase.DEProblem,t,data,priors = nothing;alg=:
                             num_samples=1000, num_warmup=1000, reltol=1e-3,
                             abstol=1e-6, maxiter=Int(1e5),likelihood=Normal,
                             vars=(StanODEData(),InverseGamma(3,3)),nchains=1,
-                            sample_u0 = false, save_idxs = nothing, diffeq_string = nothing)
+                            sample_u0 = false, save_idxs = nothing, diffeq_string = nothing, printsummary = false, summary = true)
```
Comment (Vaibhavdixit02) [inline src/stan_inference.jl:53]: The kwargs don't seem to be working, did you test it?

**Comment by Vaibhavdixit02** [comment]
We need `save_idxs` #144 for the hidden states.

File: src/turing_inference.jl [hunk 2]
```diff
@@ -15,7 +15,8 @@ function turing_inference(
     likelihood = (u,p,t,σ) -> MvNormal(u, σ[1]*ones(length(u))),
     num_samples=1000, sampler = Turing.NUTS(0.65),
     syms = [Turing.@varname(theta[i]) for i in 1:length(priors)],
-    sample_u0 = false, 
+    sample_u0 = false,
+    save_idxs = nothing,
```
Comment (ChrisRackauckas) [inline src/turing_inference.jl:19]: why was this added? It'll just work in the splatted kwargs.

File: src/stan_inference.jl [hunk 3]
```diff
@@ -50,7 +50,8 @@ function stan_inference(prob::DiffEqBase.DEProblem,t,data,priors = nothing;alg=:
                             num_samples=1000, num_warmup=1000, reltol=1e-3,
                             abstol=1e-6, maxiter=Int(1e5),likelihood=Normal,
                             vars=(StanODEData(),InverseGamma(3,3)),nchains=1,
-                            sample_u0 = false, save_idxs = nothing, diffeq_string = nothing)
+                            sample_u0 = false, save_idxs = nothing, diffeq_string = nothing, printsummary = true)
+  save_idxs !== nothing && length(save_idxs) == 1 ? save_idxs = save_idxs[1] : save_idxs = save_idxs
```
Comment (ChrisRackauckas) [inline src/stan_inference.jl:54]: ?

File: src/abc_inference.jl [hunk 4]
```diff
@@ -1,7 +1,7 @@
-function createabcfunction(prob, t, distancefunction, alg; kwargs...)
+function createabcfunction(prob, t, distancefunction, alg; save_idxs = nothing, kwargs...)
```
Comment (ChrisRackauckas) [inline src/abc_inference.jl:1]: not necessary save_idxs handling

File: README.md [hunk 5]
```diff
@@ -109,20 +110,36 @@ equation solver.
...
+ bayesian_result_stan = stan_inference(prob1,t,data,priors)
+
+ bayesian_result_turing = turing_inference(prob1,Tsit5(),t,data,priors)
+ 
+ using DynamicHMC
+ bayesian_result_hmc = dynamichmc_inference(prob1, Tsit5(), t, data, priors)
+
+ bayesian_result_abc = abc_inference(prob1, Tsit5(), t, data, priors)
+```
+### Using save_idxs for using latent variables
```
Comment (ChrisRackauckas) [inline README.md:129]: Using save_idxs for using latent variables  -> Using save_idxs to declare observables?

File: src/stan_inference.jl (hunk 3 above)
Comment (abhigupta768) [reply src/stan_inference.jl:54]: Needed to handle single valued arrays https://github.com/SciML/DiffEqBayes.jl/pull/145/files/27e634d965df3634e668f0f9e6d6787b3c135e44#diff-53adadc90178dc5b19921124ef5eefa3R81 here
